WINE DATASET

In [133]:
# Importing Wine dataset and MLPClassifier directly from sklearn

from sklearn.neural_network import MLPClassifier
import numpy as np
from sklearn.datasets import load_wine   
wine=load_wine()

In [134]:
# splitting the data set using train_test_split and keeping randomm state as birthdate

from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(wine.data, wine.target, random_state=1806)

In [135]:
# Using MLPClassifer with default parameters here to get the accuracy and error rate also using cross_val_score

from sklearn.model_selection import cross_val_score

mlp = MLPClassifier(solver='lbfgs', activation='tanh', random_state=1806,
                    
hidden_layer_sizes=[10])

mlp.fit(X_train,y_train)

print(cross_val_score(mlp, X_train, y_train,cv=5))  

scores = cross_val_score(mlp, X_train, y_train, cv=5)

gen_accuracy = sum(scr)/len(scr)

print("The generalization accuracy is:", gen_accuracy)

print("Test error rate:", 1-gen_accuracy)

[0.37037037 0.62962963 0.37037037 0.42307692 0.65384615]
The generalization accuracy is: 0.7002849002849002
Test error rate: 0.2997150997150998


In [136]:
# Using MLPClassifer with some diffferent parameters here to get the accuracy and error rate also using cross_val_score


from sklearn.model_selection import cross_val_score

mlp = MLPClassifier(solver='sgd', activation='tanh',random_state=1806,
                    
hidden_layer_sizes=[10]).fit(X_train,y_train)

print(cross_val_score(mlp, X_train, y_train,cv=5))  

scr = cross_val_score(mlp, X_train, y_train, cv=5)

gen_accuracy = sum(scr)/len(scr)

print("The generalization accuracy is:", gen_accuracy)

print("Test error rate:", 1-gen_accuracy)

[0.37037037 0.37037037 0.37037037 0.34615385 0.69230769]
The generalization accuracy is: 0.42991452991453
Test error rate: 0.5700854700854701


In [137]:
# Using MLPClassifer with different parameters here to get the accuracy and error rate also using cross_val_score


from sklearn.model_selection import cross_val_score

mlp = MLPClassifier(solver='adam', activation='tanh', random_state=1806,
                    
hidden_layer_sizes=[10]).fit(X_train,y_train)

print(cross_val_score(mlp, X_train, y_train,cv=5))  

scr = cross_val_score(mlp, X_train, y_train, cv=5)

gen_accuracy = sum(scr)/len(scr)

print("The generalization accuracy is:", gen_accuracy)

print("Test error rate:", 1-gen_accuracy)

C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, Conv

[0.37037037 0.40740741 0.48148148 0.57692308 0.42307692]
The generalization accuracy is: 0.45185185185185184
Test error rate: 0.5481481481481482


C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)


In [138]:
# Importing all the Normalisation functions from the sklearn

from sklearn.preprocessing import MinMaxScaler
from sklearn.preprocessing import StandardScaler
from sklearn.preprocessing import RobustScaler
from sklearn.preprocessing import Normalizer
from sklearn.pipeline import Pipeline
from sklearn.pipeline import make_pipeline
from sklearn.model_selection import GridSearchCV

# Pipeline Creation for different normalization techniques like MinMaxScaler,StandardScaler,RobustScaler,Normalizer

pipe_creation_1 = make_pipeline(MinMaxScaler(), mlp)
pipe_creation_2 = make_pipeline(StandardScaler(), mlp)
pipe_creation_3 = make_pipeline(RobustScaler(), mlp)
pipe_creation_4 = make_pipeline(Normalizer(), mlp)   
param_grid ={'mlpclassifier__alpha' : [0.01, 0.1, 1, 10, 100]}
grid_creation_1 = GridSearchCV(pipe_creation_1, param_grid=param_grid, cv=5, n_jobs = -1)
grid_creation_2 = GridSearchCV(pipe_creation_2, param_grid=param_grid, cv=5)
grid_creation_3 = GridSearchCV(pipe_creation_3, param_grid=param_grid, cv=5)
grid_creation_4 = GridSearchCV(pipe_creation_4, param_grid=param_grid, cv=5)

In [139]:
# fitting the gridsearch CV on the training sets to predicting the test tables

grid_creation_1.fit(X_train, y_train)
grid_creation_2.fit(X_train, y_train)
grid_creation_3.fit(X_train, y_train)
grid_creation_4.fit(X_train, y_train)
print("MinMaxScaler:", grid_creation_1.score(X_test, y_test))
print("StandardScaler:", grid_creation_2.score(X_test, y_test))
print("RobustScaler", grid_creation_3.score(X_test, y_test))
print("Normalizer", grid_creation_4.score(X_test, y_test))

C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, Conv

C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, Conv

C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, Conv

MinMaxScaler: 0.9333333333333333
StandardScaler: 0.9777777777777777
RobustScaler 0.9777777777777777
Normalizer 0.4444444444444444


C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)


USPS DATASET

In [141]:
# Using the USPS dataset and loading it using the genfromtxt

X=np.genfromtxt("test.txt",delimiter="",usecols=np.arange(0,257))

y=np.genfromtxt("train.txt",delimiter="",usecols=np.arange(0,257))

In [142]:
# Concatinating the X and Y co-ordinates of the dataset and setting randomstate as birthdate

USPS_data=np.concatenate((X[:,1:257],y[:,1:257]))

USPS_target=np.concatenate((X[:,0],y[:,0]))

X_train,X_test,y_train,y_test=train_test_split(USPS_data,USPS_target,random_state=1806)

In [143]:
# Using MLPClassifier here to get the generalised accuracy and error rate that too with default parameters,
# later I have changed the parameters and noticed the changes in values

from sklearn.model_selection import cross_val_score

mlp = MLPClassifier(solver='lbfgs', activation='tanh', random_state=1806,
                    
hidden_layer_sizes=[10])

mlp.fit(X_train,y_train)

print(cross_val_score(mlp, X_train, y_train,cv=5))  

scr = cross_val_score(mlp, X_train, y_train, cv=5)

gen_accuracy = sum(scr)/len(scr)

print("The generalization accuracy is:", gen_accuracy)

print('Test error rate:', 1-gen_accuracy)

C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:470: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
  self.n_iter_ = _check_optimize_result("lbfgs", opt_res, self.max_iter)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:470: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
  self.n_iter_ = _check_optimize_result("lbfgs", opt_res, self.max_iter)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:470: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITE

[0.91326165 0.91039427 0.91612903 0.92324247 0.91032999]


C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:470: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
  self.n_iter_ = _check_optimize_result("lbfgs", opt_res, self.max_iter)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:470: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
  self.n_iter_ = _check_optimize_result("lbfgs", opt_res, self.max_iter)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:470: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITE

The generalization accuracy is: 0.9146714799216303
Test error rate: 0.08532852007836966


C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:470: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
  self.n_iter_ = _check_optimize_result("lbfgs", opt_res, self.max_iter)


In [144]:
# Using MLPClassifier here to get accuracy and error rate but changed some parameters

from sklearn.model_selection import cross_val_score

mlp = MLPClassifier(solver='sgd', activation='tanh', random_state=1806,
                    
hidden_layer_sizes=[10]).fit(X_train,y_train)

print(cross_val_score(mlp, X_train, y_train,cv=5))  

scr = cross_val_score(mlp, X_train, y_train, cv=5)

gen_accuracy = sum(scr)/len(scr)

print("The generalization accuracy is:", gen_accuracy)

print('Test error rate:', 1-gen_accuracy)

C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, Conv

[0.93189964 0.93476703 0.92903226 0.93758967 0.92611191]


C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, Conv

The generalization accuracy is: 0.9318801005846871
Test error rate: 0.06811989941531293


C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)


In [145]:
# Importing all the Normalisation functions from the sklearn

from sklearn.preprocessing import MinMaxScaler
from sklearn.preprocessing import StandardScaler
from sklearn.preprocessing import RobustScaler
from sklearn.preprocessing import Normalizer
from sklearn.pipeline import Pipeline
from sklearn.pipeline import make_pipeline
from sklearn.model_selection import GridSearchCV

# Pipeline Creation for different normalization techniques like MinMaxScaler,StandardScaler,RobustScaler,Normalizer

pipe_creation_1 = make_pipeline(MinMaxScaler(), mlp)
pipe_creation_2 = make_pipeline(StandardScaler(), mlp)
pipe_creation_3 = make_pipeline(RobustScaler(), mlp)
pipe_creation_4 = make_pipeline(Normalizer(), mlp)
    

param_grid ={'mlpclassifier__alpha' : [0.01, 0.1, 1, 10, 100]} #Using Alpha wiht default values for the grid

grid_creation_1 = GridSearchCV(pipe, param_grid=param_grid, cv=5, n_jobs = -1)
grid_creation_2 = GridSearchCV(pipe2, param_grid=param_grid, cv=5)
grid_creation_3 = GridSearchCV(pipe3, param_grid=param_grid, cv=5)
grid_creation_4 = GridSearchCV(pipe4, param_grid=param_grid, cv=5)

In [146]:
# fitting the gridsearch CV on the training sets to predicting the test tables

grid_creation_1.fit(X_train, y_train)
grid_creation_2.fit(X_train, y_train)
grid_creation_3.fit(X_train, y_train)
grid_creation_4.fit(X_train, y_train)

print("MinMaxScaler:", grid_creation_1.score(X_test, y_test))
print("StandardScaler:", grid_creation_2.score(X_test, y_test))
print("RobustScaler", grid_creation_3.score(X_test, y_test))
print("Normalizer", grid_creation_4.score(X_test, y_test))

C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, Conv

C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)
C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, Conv

MinMaxScaler: 0.9264516129032258
StandardScaler: 0.9234408602150538
RobustScaler 0.9260215053763441
Normalizer 0.843010752688172


C:\Users\saura\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:571: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  % self.max_iter, ConvergenceWarning)


SVM for the WINE Dataset.

In [147]:
#Importing and loading SVC and Wine dataset from sklearn

from sklearn.svm import SVC
from sklearn.datasets import load_wine
wine=load_wine()

In [148]:
# splitting dataset using random state as birthdate

X_train,X_test,y_train,y_test=train_test_split(wine.data,wine.target,random_state=1806)

In [149]:
#Using Cross validation score we are getting accuracy and error rate here

from sklearn.model_selection import cross_val_score

svm = SVC() 

svc.fit(X_train, y_train)

scr = cross_val_score(svm, X_train, y_train, cv=5)

gen_accuracy = sum(scr)/len(scr)

print("The generalization accuracy is:", gen_accuracy)

print('Test error rate:', 1-gen_accuracy)

The generalization accuracy is: 0.7002849002849002
Test error rate: 0.2997150997150998


In [150]:
# Pipeline Creation for different normalization techniques like MinMaxScaler,StandardScaler,RobustScaler,Normalizer

pipe_creation_1 = make_pipeline(MinMaxScaler(), SVC())
pipe_creation_2 = make_pipeline(StandardScaler(), SVC())
pipe_creation_3 = make_pipeline(RobustScaler(), SVC())
pipe_creation_4 = make_pipeline(Normalizer(), SVC())   
param_grid = {'svc__C': [0.01, 0.1, 1, 10, 100],'svc__gamma': [0.001, 0.01, 0.1, 1, 10, 100]}
grid_creation_1 = GridSearchCV(pipe_creation_1, param_grid=param_grid, cv=5)
grid_creation_2 = GridSearchCV(pipe_creation_2, param_grid=param_grid, cv=5)
grid_creation_3 = GridSearchCV(pipe_creation_3, param_grid=param_grid, cv=5)
grid_creation_4 = GridSearchCV(pipe_creation_4, param_grid=param_grid, cv=5)

In [151]:
# fitting the gridsearch CV on the training sets to predicting the test tables

grid_creation_1.fit(X_train, y_train)
grid_creation_2.fit(X_train, y_train)
grid_creation_3.fit(X_train, y_train)
grid_creation_4.fit(X_train, y_train)

print("MinMaxScaler:", grid_creation_1.score(X_test, y_test))
print("StandardScaler:", grid_creation_2.score(X_test, y_test))
print("RobustScaler", grid_creation_3.score(X_test, y_test))
print("Normalizer", grid_creation_4.score(X_test, y_test))


MinMaxScaler: 1.0
StandardScaler: 0.9777777777777777
RobustScaler 1.0
Normalizer 0.9111111111111111


SVM on USPS Dataset

In [152]:
# Importing and loading SVC and USPS dataset from sklearn

from sklearn.svm import SVC

X=np.genfromtxt("test.txt",delimiter="",usecols=np.arange(0,257))

y=np.genfromtxt("train.txt",delimiter="",usecols=np.arange(0,257))

In [153]:
# Concatinating the X and Y co-ordinates of the dataset and setting randomstate as birthdate

USPS_data=np.concatenate((X[:,1:257],y[:,1:257]))

USPS_target=np.concatenate((X[:,0],y[:,0]))

X_train,X_test,y_train,y_test=train_test_split(USPS_data,USPS_target,random_state=1806)

In [154]:
# Using Cross validation score we are getting accuracy and error rate here


from sklearn.model_selection import cross_val_score

svm = SVC() 

svc.fit(X_train, y_train)

scr = cross_val_score(svm, X_train, y_train, cv=5)

gen_accuracy = sum(scr)/len(scr)

print("The generalization accuracy is:", gen_accuracy)

print('Test error rate:', 1-gen_accuracy)

The generalization accuracy is: 0.9720350915084103
Test error rate: 0.027964908491589724


In [160]:
# Pipeline Creation for different normalization techniques like MinMaxScaler,StandardScaler,RobustScaler,Normalizer


pipe_creation_1 = make_pipeline(MinMaxScaler(), SVC())
pipe_creation_2 = make_pipeline(StandardScaler(), SVC())
pipe_creation_3 = make_pipeline(RobustScaler(), SVC())
pipe_creation_4 = make_pipeline(Normalizer(), SVC())   
param_grid = {'svc__C': [0.01, 0.1, 1, 10, 100],'svc__gamma': [0.001, 0.01, 0.1, 1, 10, 100]}
grid_creation_1 = GridSearchCV(pipe_creation_1, param_grid=param_grid, cv=5, n_jobs = -1)
grid_creation_2 = GridSearchCV(pipe_creation_2, param_grid=param_grid, cv=5, n_jobs = -1)
grid_creation_3 = GridSearchCV(pipe_creation_3, param_grid=param_grid, cv=5, n_jobs= -1)
grid_creation_4 = GridSearchCV(pipe_creation_4, param_grid=param_grid, cv=5, n_jobs = -1)

In [ ]:
# fitting the gridsearch CV on the training sets to predicting the test tables

grid_creation_1.fit(X_train, y_train)
grid_creation_2.fit(X_train, y_train)
grid_creation_3.fit(X_train, y_train)
grid_creation_4.fit(X_train, y_train)

print("MinMaxScaler:", grid_creation_1.score(X_test, y_test))
print("StandardScaler:", grid_creation_2.score(X_test, y_test))
print("RobustScaler", grid_creation_3.score(X_test, y_test))
print("Normalizer", grid_creation_4.score(X_test, y_test

All Numerical-Result Values

(A)Using MLPClassifer for WINE dataset:

1.The generalization accuracy is: 0.7002849002849002 (using default parameters)
  Test error rate: 0.2997150997150998 (using default parameters)
  
2.The generalization accuracy is: 0.42991452991453
  Test error rate: 0.5700854700854701  
  
3.The generalization accuracy is: 0.45185185185185184
  Test error rate: 0.5481481481481482 

4.Test labels for different normalisation techniques for WINE dataset
  MinMaxScaler:  0.9333333333333333
  StandardScaler:0.9777777777777777
  RobustScaler:   0.9777777777777777
  Normalizer:     0.4444444444444444

(B)Using MLPClassifer for USPS dataset:

1.The generalization accuracy is: 0.9146714799216303 (using default parameters)
  Test error rate: 0.08532852007836966 (using default parameters)

2.The generalization accuracy is: 0.9318801005846871
  Test error rate: 0.06811989941531293

3.Test labels for different normalisation techniques for USPS dataset
  MinMaxScaler: 0.9264516129032258
  StandardScaler: 0.9234408602150538
  RobustScaler 0.9260215053763441
  Normalizer 0.843010752688172
  
(C)SVM on WINE Dataset:

1.The generalization accuracy is: 0.7002849002849002
  Test error rate: 0.2997150997150998
 
2.Test labels for different normalisation techniques for WINE dataset
  MinMaxScaler: 1.0
  StandardScaler: 0.9777777777777777
  RobustScaler 1.0
  Normalizer 0.9111111111111111 
  
(D) SVM on USPS Dataset:

1.The generalization accuracy is: 0.9720350915084103
  Test error rate: 0.027964908491589724
  
2.Test labels for different normalisation techniques for USPS dataset 
